parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoWriteSupport.java

```diff
@@ -506,15 +506,13 @@ private void validatedMapping(Descriptor descriptor, GroupType parquetSchema) {
   class TimestampWriter extends FieldWriter {
     @Override
     final void writeRawValue(Object entry) {
-      List<Object> timeFields = ((MessageOrBuilder) entry).getAllFields().values().stream().collect(Collectors.toList());
-      if(timeFields.size() == 0) {
-        recordConsumer.addLong(0L);
-        return;
-      }
-      long eventTimestampMillis = (((Long) timeFields.get(0)) * 1000);
-      if(timeFields.size() > 1) {
-        eventTimestampMillis += (((Integer)(timeFields.get(1))) / 1000000);
+      final Timestamp timestamp;
```

Comment:
What if we do this to avoid serialize/deserialize and try/catch ?
```
class TimestampWriter extends FieldWriter {
    Descriptors.FieldDescriptor secondFD, nanosFD;
    TimestampWriter() {
      secondFD = com.google.protobuf.Timestamp.getDescriptor().findFieldByNumber(1);
      nanosFD = com.google.protobuf.Timestamp.getDescriptor().findFieldByNumber(2);
    }
    @Override
    final void writeRawValue(Object entry) {
      MessageOrBuilder message = ((MessageOrBuilder) entry);
      long seconds = (long) message.getField(secondFD);
      int nanos = (int) message.getField(nanosFD);
      long eventTimestampMillis = seconds * 1000 + nanos/1000000;
      recordConsumer.addLong(eventTimestampMillis);
    }
}
```